tangelo/linq/translator/translate_stim.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2021 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions helping with quantum circuit and operator format conversion between
+Tangelo format and qiskit format.
+
+In order to produce an equivalent circuit for the target backend, it is
+necessary to account for:
+- how the gate names differ between the source backend to the target backend.
+- how the order and conventions for some of the inputs to the gate operations
+    may also differ.
+"""
+
+from tangelo.linq import Circuit
+from tangelo.helpers import deprecated
+from tangelo.linq.helpers.circuits.clifford_circuits import decompose_gate_to_cliffords
+
+
+def get_stim_gates():
+    """Map gate name of the Tangelo format to the equivalent add_gate method of
+    Stim's CircuitInstruction class API and supported gates:
+    https://github.com/quantumlib/Stim/blob/main/doc/gates.md
+    """
+    GATE_STIM = dict()
+    GATE_STIM["I"] = "I"
+    GATE_STIM["H"] = "H"
+    GATE_STIM["X"] = "X"
+    GATE_STIM["Y"] = "Y"
+    GATE_STIM["Z"] = "Z"
+    GATE_STIM["S"] = "S"
+    GATE_STIM["SDAG"] = "S_DAG"
+    GATE_STIM["CX"] = "CX"
+    GATE_STIM["CY"] = "CY"
+    GATE_STIM["CZ"] = "CZ"
+    GATE_STIM["CNOT"] = "CX"
+    GATE_STIM["SWAP"] = "SWAP"
+    GATE_STIM["MEASURE"] = "M"
+    GATE_STIM["R"] = "R"
+    return GATE_STIM
+
+
+def direct_tableau(source_circuit):
```

Comment:
The name here is not super descriptive. Why not just translate_tableau ?

I'm also confused about what this does. Your doc is not correct, we are not returning a `stim.QuantumCircuit`. Your code returns a `stim.TableauSimulator` but it seems to be a circuit of some sort ?